lib/ohai/plugins/solaris2/network.rb

```diff
@@ -104,7 +104,9 @@ def full_interface_name(iface, part_name, index)
 
     so.stdout.lines do |line|
       # regex: http://rubular.com/r/Iag7JLVTVe
-      if line =~ /^([0-9a-zA-Z\.\:\-]+): \S+ mtu (\d+) index (\d+)/
+      if line =~ /^([0-9a-zA-Z\.\:\-]+): \S+ mtu (\d+) index (\d+)/ ||
+          # regex: https://rubular.com/r/0vSO7KocyLBHTd
+          line =~ /^([0-9a-zA-Z\.\:\-]+): \S+ mtu (\d+)/
```

Comment:
These two can be collapsed into one: `/^([0-9a-zA-Z\.\:\-]+): \S+ mtu (\d+)(?: index (\d+))?/`

Reply:
Nice, thanks for this